src/posix/platform/netif.cpp

```diff
@@ -1636,11 +1636,22 @@ static void processNetlinkEvent(otInstance *aInstance)
 
     length = recv(sNetlinkFd, msgBuffer.buffer, sizeof(msgBuffer.buffer), 0);
 
-    VerifyOrExit(length > 0);
+#if defined(__linux__)
+#define HEADER_SIZE sizeof(nlmsghdr)
+#else
+#define HEADER_SIZE sizeof(rt_msghdr)
+#endif
+
+    // Ensures full netlink header is received
+    if (length < static_cast<ssize_t>(HEADER_SIZE))
+    {
+        otLogWarnPlat("[netif] Unexpected netlink recv() result: %ld", static_cast<long>(length));
+        ExitNow();
+    }
 
 #if defined(__linux__)
-    for (struct nlmsghdr *msg = &msgBuffer.nlMsg; NLMSG_OK(msg, static_cast<size_t>(length));
-         msg                  = NLMSG_NEXT(msg, length))
+    for (struct nlmsghdr *msg                                                               = &msgBuffer.nlMsg;
+         NLMSG_OK(msg, static_cast<size_t>(length)) && (msg->nlmsg_type != NLMSG_DONE); msg = NLMSG_NEXT(msg, length))
```

Comment:
perhaps we just include the NLMSG_DONE in the `switch` statement below.

Reply:
done

Reply:
Good suggestion. 
I was looking at the `NLMSG_NEXT()` doc and saw it is suggestion to exit on `DONE`.